webui/react/src/pages/TaskList.tsx

```diff
@@ -108,6 +113,66 @@ const TaskList: React.FC = () => {
     triggerTensorboardsRequest,
   ]);
 
+  const handleSourceExpand = useCallback((event: React.MouseEvent, id: string) => {
+    event.stopPropagation();
+    setSourceExpanded(prev => {
+      const newSourceExpanded = { ...prev };
+      if (!newSourceExpanded[id]) newSourceExpanded[id] = true;
+      else delete newSourceExpanded[id];
+      return newSourceExpanded;
+    });
+  }, []);
+
+  const columns = useMemo(() => {
+    const sourceRenderer: TaskRenderer = (_, record) => {
+      const info = {
+        isPlural: false,
+        label: '',
+        path: '',
+        source: [] as number[],
+      };
+      if (record.misc?.experimentIds) {
+        info.label = 'Experiment';
+        info.path = '/det/experiments';
+        info.source = record.misc.experimentIds || [];
+      } else if (record.misc?.trialIds) {
+        info.label = 'Trial';
+        info.path = '/ui/trials';
+        info.source = record.misc.trialIds || [];
+      }
+      info.isPlural = info.source.length > 1;
+      info.source.sort(numericSorter);
+
+      const isExpanded = sourceExpanded[record.id];
+      const sourceCount = info.source.length;
+      const showToggle = info.source.length > MAX_SOURCES;
+      const toggleLabel = `${isExpanded ? 'Hide' : 'Show'} ${sourceCount}${isExpanded ? '' : '+'}`;
+
+      return sourceCount !== 0 ? (
+        <div className={css.sourceLinks}>
+          <label>{info.label}{info.isPlural ? 's' : ''}</label>
+          {info.source.map((id, index) => {
+            const display = index < MAX_SOURCES || isExpanded ? 'inline' : 'none';
+            const handleClick = (event: React.MouseEvent) => {
+              event.stopPropagation();
+              makeClickHandler(`${info.path}/${id}`)(event);
```

Comment:
The link to `/det/experiments/$ID` leads to a page that doesn't render correctly. I'm assuming this is because the experiment detail page isn't finished yet and is unrelated to this PR?

Reply:
ah yes, I should point to `/ui/experiments` instead until we turn on experiments detail. thanks!